dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/blas/jblas/MatrixJBLASImpl.java

```diff
@@ -0,0 +1,485 @@
+    matrix.divi(((MatrixJBLASImpl) m).matrix);
+    return this;
+  }
+
+  @Override
+  public Matrix divColumnVector(final Matrix v) {
+    checkImpl(v);
+    return new MatrixJBLASImpl(matrix.divColumnVector(((MatrixJBLASImpl) v).matrix));
+  }
+
+  @Override
+  public Matrix diviColumnVector(final Matrix v) {
+    checkImpl(v);
+    matrix.diviColumnVector(((MatrixJBLASImpl) v).matrix);
+    return this;
+  }
+
+  @Override
+  public Matrix divRowVector(final Matrix v) {
+    checkImpl(v);
+    return new MatrixJBLASImpl(matrix.divRowVector(((MatrixJBLASImpl) v).matrix));
+  }
+
+  @Override
+  public Matrix diviRowVector(final Matrix v) {
+    checkImpl(v);
+    matrix.diviRowVector(((MatrixJBLASImpl) v).matrix);
+    return this;
+  }
+
+  @Override
+  public Matrix rdiv(final float v) {
+    return new MatrixJBLASImpl(matrix.rdiv(v));
+  }
+
+  @Override
+  public Matrix rdivi(final float v) {
+    matrix.rdivi(v);
+    return this;
+  }
+
+  @Override
+  public Matrix rdiv(final Matrix m) {
+    checkImpl(m);
+    return new MatrixJBLASImpl(matrix.rdiv(((MatrixJBLASImpl) m).matrix));
+  }
+
+  @Override
+  public Matrix rdivi(final Matrix m) {
+    checkImpl(m);
+    matrix.rdivi(((MatrixJBLASImpl) m).matrix);
+    return this;
+  }
+
+  @Override
+  public Matrix mmul(final Matrix m) {
+    checkImpl(m);
+    return new MatrixJBLASImpl(matrix.mmul(((MatrixJBLASImpl) m).matrix));
+  }
+
+  @Override
+  public Matrix mmuli(final Matrix m) {
+    checkImpl(m);
+    matrix.mmuli(((MatrixJBLASImpl) m).matrix);
+    return this;
+  }
+
+  @Override
+  public float max() {
+    return matrix.max();
+  }
+
+  @Override
+  public Matrix columnMaxs() {
+    return new MatrixJBLASImpl(matrix.columnMaxs());
+  }
+
+  @Override
+  public Matrix rowMaxs() {
+    return new MatrixJBLASImpl(matrix.rowMaxs());
+  }
+
+  @Override
+  public float min() {
+    return matrix.min();
+  }
+
+  @Override
+  public Matrix columnMins() {
+    return new MatrixJBLASImpl(matrix.columnMins());
+  }
+
+  @Override
+  public Matrix rowMins() {
+    return new MatrixJBLASImpl(matrix.rowMins());
+  }
+
+  @Override
+  public Matrix columnSums() {
+    return new MatrixJBLASImpl(matrix.columnSums());
+  }
+
+  @Override
+  public Matrix rowSums() {
+    return new MatrixJBLASImpl(matrix.rowSums());
+  }
+
+  @Override
+  public float sum() {
+    return matrix.sum();
+  }
+
+  @Override
+  public boolean compare(final Matrix m, final float tolerance) {
+    if (m instanceof MatrixJBLASImpl) {
+      return matrix.compare(((MatrixJBLASImpl) m).matrix, tolerance);
+    }
+    return false;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (o instanceof MatrixJBLASImpl) {
+      return matrix.equals(((MatrixJBLASImpl) o).matrix);
+    }
+    return false;
+  }
+
+  @Override
+  public int hashCode() {
+    return matrix.hashCode();
+  }
+
+  public static MatrixJBLASImpl concatHorizontally(final MatrixJBLASImpl a, final MatrixJBLASImpl b) {
```

Comment:
According to the current implementation, to concat two `Matrix`s, we need to receive an injection of a `MatrixFactory` and use `MatrixFactory.concatxx()`. Couldn't we just have concatenation as a method of the `Matrix` implementation?